scripts/list_outdated_dependencies.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Check capped dependencies for newer versions."""
+import re
+import sys
+from copy import deepcopy
+from pathlib import Path
+from typing import Any
+
+import httpx
+import stringcase
+import tomli
+from packaging.requirements import Requirement
+
+from script_utils import cli
+
+REPO_ROOT_DIR = Path(__file__).parent.parent.resolve()
+PYPROJECT_TOML_PATH = REPO_ROOT_DIR / "pyproject.toml"
+DEV_DEPS_PATH = REPO_ROOT_DIR / "requirements-dev.in"
+LOCK_FILE_PATH = REPO_ROOT_DIR / "requirements-dev.txt"
+
+
+def get_transitive_deps(exclude: set[str]) -> list[str]:
+    """Inspect the lock file to get the transitive dependencies"""
+    dependency_pattern = re.compile(r"([^=\s]+==[^\s]*?)\s")
```

Comment:
In [6ca45d6](https://github.com/ghga-de/microservice-repository-template/pull/165/commits/6ca45d6feec751c223a410149455496f2045ad30), I moved the function in question to `lock_deps.py`. The reason I put it in a separate file was to prevent having to include `stringcase` in the pre-commit-hook config when the local hook doesn't actually use it. 
The other script that uses this function, `update_hook_revs`, has a slightly different use case and the list of raw strings would not be useful, so it returns Requirement objects now.